packages/common/src/hooks/chats/useAudiusLinkResolver.ts

```diff
@@ -89,20 +96,23 @@ export const useAudiusLinkResolver = ({
 }) => {
   // Maintain bidirectional maps of audius links to human readable format
   const linkToHuman = useRef<{ [key: string]: string }>({}).current
-  const humanToData = useRef<{
-    [key: string]: { link: string; data: Track | Playlist | User }
-  }>({}).current
+  const humanToData = useRef<{ [key: string]: LinkEntity }>({}).current
 
-  // The track id used to render the composer preview
-  const [trackId, setTrackId] = useState<Nullable<ID>>(null)
-  // The collection id used to render the composer preview
-  const [collectionId, setCollectionId] = useState<Nullable<ID>>(null)
+  const getSortedMatchStrings = useCallback(() => {
+    return Object.keys(humanToData).sort((a, b) => b.length - a.length)
+  }, [humanToData])
+
+  const getSortedRegex = useCallback(() => {
+    return getSortedMatchStrings()
+      .map((str) => escapeRegexSpecialChars(str))
+      .join('|')
+  }, [getSortedMatchStrings])
 
   /**
    * Resolves Audius links in the provided text to their display text equivalent.
    */
   const resolveLinks = useCallback(
-    async (value: string) => {
+    async (value: string, _cursorPosition = 0) => {
```

Comment:
can we docstring what _cursorPosition is for?

Reply:
was going to use it to better preserve the caret position when unfurling a link/autocompleting, but didn't get around to it yet. I'll remove for now